src/migration.ts

```diff
@@ -0,0 +1,269 @@
+import {Comorbidity} from "./models/comorbidity.js";
+import {Biomarker} from "./models/biomarker.js";
+import {Donor} from "./models/donor.js";
+import {Exposure} from "./models/exposure.js";
+import {FamilyHistory} from "./models/family-history.js";
+import {FollowUp} from "./models/follow-up.js";
+import {PrimaryDiagnosis} from "./models/primary-diagnosis.js";
+import {Treatment} from "./models/treatment.js";
+import {Specimen} from "./models/specimen.js";
+import {SampleRegistration} from "./models/sample-registration.js";
+import * as fs from "fs";
+import {Response} from "express";
+
+var myLogFileStream = fs.createWriteStream("./console.log");
+var myConsole = new console.Console(myLogFileStream, myLogFileStream);
+
+let checkpoint = '';
+export async function beginMigration(response: Response) {
+
+    console.log("");
+    console.log("");
+    console.log("The Great Wildebeest Migration begins....");
+    console.log("");
+    console.log("...3");
+    console.log("......2");
+    console.log(".........1");
+    console.log("............ GO!");
+    console.log("");
+
+    await MongoDataSource.initialize();
+    await  PostgresDataSource.initialize();
+
+    const cdRepo = MongoDataSource.getRepository(ClinicalDonor);
+    const donors = await cdRepo.find();
+
+    for (const donor of donors) {
+        console.log("");
+        console.log(" ------------ Migrating ids for  donor: " + donor.id+" Program Id: "+donor.programId+" -Submitter Id: "+donor.submitterId+" -------------");
+        console.log("");
+        try {
+            await migrateDonors(donor);
+            await migrateBiomarkers(donor);
+            await migrateComorbidity(donor);
+            await migrateExposure(donor);
+            await migrateFamilyHistory(donor);
+            await migrateFollowUps(donor);
+            await migratePrimaryDiagnosis(donor);
+            await migrateTreatments(donor);
+            await migrateSpecimens(donor);
+            await migrateSamples(donor);
+        } catch (e) {
+            console.log("Oops, Croc attack! - migration for donor failed."+" -Checkpoint: "+checkpoint+" -Program Id: "+donor.programId+" -Submitter Id: "+donor.submitterId);
+            console.log("Problem: "+e.message);
+            await MongoDataSource.getRepository(FailedMigrations).save(donor);
+            continue;
+        }
+    }
+
+    console.log("");
+    console.log('Wildebeest migrated!');
+    response.send;
+}
+
+
+async function migrateDonors(donor: ClinicalDonor){
+    checkpoint = 'Donor: '+donor.donorId;
+
+    const dnRepo = PostgresDataSource.getRepository(Donor);
+    const submitterDonorId = donor.submitterId?.toString();
+    const programId = donor.programId;
+    const idDonor = new Donor();
+    idDonor.submitterDonorId = submitterDonorId || '-';
+    idDonor.programId = programId || '-';
+    idDonor.entityId = donor.donorId?.toString() || '-';
+    idDonor.entityType = "donor";
```

Comment:
Might want to [add constructors to your classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) and refactor like this:

```
const emptyValue = '-';
const submitterDonorId = donor.submitterId?.toString() || emptyValue;
const programId = donor.programId || emptyValue;
const entityId = donor.donorId?.toString() || emptyValue;
const entityType = "donor";

const idDonor = new Donor({ 
    entityType,
    programId,
    entityId,
    submitterDonorId,
});
```

Unless there's any reason this won't work with Type ORM

Reply:
They should work with TypeOrm. I am just keeping them this way because it is easy for me to troubleshoot if something goes wrong.